api/envoy/api/v2/core/protocol.proto

```diff
@@ -42,6 +42,24 @@ message Http1ProtocolOptions {
   // Envoy does not otherwise support HTTP/1.0 without a Host header.
   // This is a no-op if *accept_http_10* is not true.
   string default_host_for_http_10 = 3;
+
+  message HeaderKeyFormat {
+    enum HeaderFormat {
```

Comment:
enums are pretty fragile for stuff like this. I would recommend just making this a oneof with a required value and a single `TrainCase` message that is empty. In the future we can easily add configuration to the TrainCase message or add more complicated message types.

Reply:
That's probably a good idea.